cpp/src/packed/chunk_manager.cpp

```diff
@@ -60,12 +57,19 @@ std::vector<std::shared_ptr<arrow::ArrayData>> ChunkManager::SliceChunks(
       chunk_state.addChunk(1);
       chunk_state.resetOffset();
       slice_data = (offset > 0) ? chunk->Slice(offset, chunksize_)->data() : chunk->data();
+      if (chunk_state.chunk == chunk_sizes[i]) {
+        table_to_pop.insert(column_offsets_[i].path_index);
```

Comment:
Why can't we just pop the tables in place?

Reply:
the tables are from row groups that needed to be read. If the chunk achieves the chunk size, there is no more chunk in the table to be read. Therefore we can pop out the table from the queue. In the next run, it will process the new table.